src/ICSharpCode.SharpZipLib/Zip/ZipStrings.cs

```diff
@@ -20,175 +25,100 @@ static ZipStrings()
 			{
 				SystemDefaultCodePage = FallbackCodePage;
 			}
-		}
-
-		/// <summary>Code page backing field</summary>
-		/// <remarks>
-		/// The original Zip specification (https://pkware.cachefly.net/webdocs/casestudies/APPNOTE.TXT) states
-		/// that file names should only be encoded with IBM Code Page 437 or UTF-8.
-		/// In practice, most zip apps use OEM or system encoding (typically cp437 on Windows).
-		/// Let's be good citizens and default to UTF-8 http://utf8everywhere.org/
-		/// </remarks>
-		private static int codePage = AutomaticCodePage;
 
-		/// Automatically select codepage while opening archive
-		/// see https://github.com/icsharpcode/SharpZipLib/pull/280#issuecomment-433608324
-		/// 
-		private const int AutomaticCodePage = -1;
+			SystemDefaultEncoding = Encoding.GetEncoding(SystemDefaultCodePage);
+		}
 
 		/// <summary>
-		/// Encoding used for string conversion. Setting this to 65001 (UTF-8) will
-		/// also set the Language encoding flag to indicate UTF-8 encoded file names.
+		/// If set, use the encoding set by <see cref="CodePage"/> for zip entries instead of the defaults
 		/// </summary>
-		public static int CodePage
-		{
-			get
-			{
-				return codePage == AutomaticCodePage? Encoding.UTF8.CodePage:codePage;
-			}
-			set
-			{
-				if ((value < 0) || (value > 65535) ||
-					(value == 1) || (value == 2) || (value == 3) || (value == 42))
-				{
-					throw new ArgumentOutOfRangeException(nameof(value));
-				}
+		public bool ForceZipLegacyEncoding { get; set; }
 
-				codePage = value;
-			}
-		}
+		/// <summary>
+		/// The default encoding used for ZipCrypto passwords in zip files, set to <see cref="SystemDefaultEncoding"/>
```

Comment:
Is that something that might differ between .NET 4.x and .NETCore on Windows? (might be UTF-8 on Core, something else on 4.8?)

Reply:
Yeah, https://docs.microsoft.com/en-us/dotnet/api/system.text.encoding.default?view=net-5.0#the-default-property-on-net-core 

Does it matter though? It still seems like the best defaults for non-windows and core-on-windows consumers... They can override the encoding if they know what to use, but we cant detect the OS defaults without interop afaik.

Reply:
>Does it matter though?
Quite possibly not (or possibly it's the only thing to use that doesn't need the extra encoding packages registered anyway).

What's the situation with using Encoding.Default vs Encoding.GetEncoding(0) on core? (i.e. if the behaviour of GetEncoding(0) might vary depending on which providers have been registered, and .Default is constant, which one is best?)

Reply:
Yeah, if the consumer registers additional providers *before* the **static** constructor of `ZipStrings` it will return another value. It's probably better to use `Encoding.Default` instead, and on windows, explicitly call [GetACP](https://docs.microsoft.com/en-us/windows/win32/api/winnls/nf-winnls-getacp).